server/src/main/java/org/elasticsearch/index/reindex/BulkByScrollTask.java

```diff
@@ -610,6 +961,41 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
             return builder;
         }
 
+        public static StatusOrException fromXContent(XContentParser parser) throws IOException {
```

Comment:
Can you leave a comment about why you do it this way? The reasoning is fairly involved and a comment will totally help future-me when I reread this six months from now.